godot_project/project_workspace/workspace_context/workspace_context.gd

```diff
@@ -833,13 +833,27 @@ func disable_hydrogens_visualization(clear_selection: bool = true) -> void:
 			# Ignore virtual objects
 			continue
 		context.nano_structure.disable_hydrogens_visibility()
-		if clear_selection:
-			context.clear_selection()
-	
+		if in_clear_hydrogen_selection:
+			_deselect_hydrogens(context)
+		
 	rendering.disable_hydrogens()
 	workspace.representation_settings.set_hydrogen_visibility_and_notify(false)
 
 
+func _deselect_hydrogens(out_structure_context: StructureContext) -> void:
+	var atoms_to_deselect: PackedInt32Array = PackedInt32Array()
+	var bonds_to_deselect: PackedInt32Array = PackedInt32Array()
+	var atoms: PackedInt32Array = out_structure_context.get_selected_atoms()
+	for atom_id: int in atoms:
+		var is_hydrogen: bool = out_structure_context.nano_structure.atom_get_atomic_number(atom_id) == \
+				PeriodicTable.ATOMIC_NUMBER_HYDROGEN
+		if is_hydrogen:
+			atoms_to_deselect.append(atom_id)
+			bonds_to_deselect.append_array(out_structure_context.nano_structure.atom_get_bonds(atom_id))
```

Comment:
this code doesn't consider bonds that are selected but the hydrogen atom is not selected